src/module.c

```diff
@@ -554,12 +557,12 @@ void jl_binding_deprecation_warning(jl_module_t *m, jl_binding_t *b)
                             jl_printf(JL_STDERR, ".");
                         }
                         jl_printf(JL_STDERR, "%s", jl_symbol_name(mt->name));
-                        jl_printf(JL_STDERR, " instead");
+                        jl_printf(JL_STDERR, " instead.");
                     }
                 }
             }
         }
-        jl_printf(JL_STDERR, ".\n");
+        jl_printf(JL_STDERR, "\n");
```

Comment:
I could be wrong, but I believe it's this line that's causing the tests to fail. They can be updated to pass with this change:
```diff
--- a/test/cmdlineargs.jl
+++ b/test/cmdlineargs.jl
@@ -233,7 +233,7 @@ let exename = `$(Base.julia_cmd()) --sysimage-native-code=yes --startup-file=no`
             close(out.in)
             wait(proc)
             @test success(proc)
-            @test wait(output) == "WARNING: Foo.Deprecated is deprecated, use NotDeprecated instead.\n  likely near no file:5"
+            @test wait(output) == "WARNING: Foo.Deprecated is deprecated, use NotDeprecated instead\n  likely near no file:5"
         end
 
         let out  = Pipe(),
```
Whether you change the implementation or the test to fix the CI failures I guess depends on how much you want that period. 🙂